File: src/forgesight/api/schemas/documents.py
"""Request/response schemas for the document/RAG search endpoints."""

from __future__ import annotations

import uuid
from datetime import datetime
from typing import Optional

from pydantic import ConfigDict
from sqlmodel import SQLModel


class RetrievedPassageResponse(SQLModel):
    passage_id: uuid.UUID
    document_id: str
    document_title: str
    document_version: str
    document_date: Optional[str] = None
    section_title: Optional[str] = None
    section_reference: Optional[str] = None
    chunk_text: str
    retrieval_score: float
    rerank_score: Optional[float] = None
    retrieval_query: str
    retrieval_timestamp: datetime
    embedding_model: str
    retrieved_by: str

    model_config = ConfigDict(from_attributes=True)


class DocumentSearchResponse(SQLModel):
    passages: list[RetrievedPassageResponse]
    no_relevant_document_found: bool
    result_count: int


class DocumentDetailResponse(SQLModel):
    document_id: str
    title: str
    category: str
    version: str
    document_date: str
    author: str
    approved_by: str
    status: str
    language: str
    full_text: Optional[str] = None


class HistoricalIncidentMatchResponse(SQLModel):
    incident_id: str
    title: str
    defect_type: str
    root_cause_confirmed: Optional[str] = None
    corrective_action_taken: Optional[str] = None
    similarity_score: float
    retrieval_timestamp: datetime


class HistoricalIncidentSearchResponse(SQLModel):
    matches: list[HistoricalIncidentMatchResponse]
    result_count: int

File: frontend/forgesight-web/src/types/documents.ts
export interface RetrievedPassageResponse {
  passage_id: string;
  document_id: string;
  document_title: string;
  document_version: string;
  document_date: string | null;
  section_title: string | null;
  section_reference: string | null;
  chunk_text: string;
  retrieval_score: number;
  rerank_score: number | null;
  retrieval_query: string;
  retrieval_timestamp: string;
  embedding_model: string;
  retrieved_by: string;
}

export interface DocumentSearchResponse {
  passages: RetrievedPassageResponse[];
  no_relevant_document_found: boolean;
  result_count: number;
}

export interface DocumentDetailResponse {
  document_id: string;
  title: string;
  category: string;
  version: string;
  document_date: string;
  author: string;
  approved_by: string;
  status: string;
  language: string;
  full_text: string | null;
}

export interface HistoricalIncidentMatchResponse {
  incident_id: string;
  title: string;
  defect_type: string;
  root_cause_confirmed: string | null;
  corrective_action_taken: string | null;
  similarity_score: number;
  retrieval_timestamp: string;
}